frontend/src/metabase/api/dashboard.ts

```diff
@@ -133,7 +133,7 @@ export const dashboardApi = Api.injectEndpoints({
       ],
     }),
     createDashboardPublicLink: builder.mutation<
-      { uuid: Dashboard["public_uuid"] },
+      Pick<Dashboard, "id"> & { uuid: Dashboard["public_uuid"] },
```

Comment:
I wonder if we can use `DashboardId` in place of `Pick<Dashboard, "id">` as we only need a single field.

Reply:
I would like to do that but we can't do something like `DashboardId & {uuid: ....}` since `DashboardId` isn't an object, so we would have to do something like `{id: DashboardId, uuid: Dashboard['public_uuid']}`. I would prefer to actually keep it like this because we can keep the association between Dashboard's id property rather than with DashboardId itself. If the definition of Dashboard `id` changes, then we get the type change for free here